src/kpi_rover_ecu/src/motor.cpp

```diff
@@ -41,9 +58,31 @@ int Motor::MotorStop() const {
     return 0;
 }
 
-int Motor::GetEncoderCounter() const {
+int Motor::GetEncoderCounter() {
     const int kEncoderTicks = rc_encoder_read(motorNumber_);
+    int pid_encoder_ticks = kEncoderTicks;
+    if (!inverted_) {
+        pid_encoder_ticks *= -1;
+    }
+
+    const auto kCurrentTimePoint = std::chrono::high_resolution_clock::now();
+    const std::chrono::duration<double, std::milli> kElapsedMilliseconds = kCurrentTimePoint - lastTimePoint_;
+    lastTimePoint_ = kCurrentTimePoint;
+    const auto kTimeDt = static_cast<float>(kElapsedMilliseconds.count());
+
+    const float kRevolutions = static_cast<float>(pid_encoder_ticks) / static_cast<float>(kLoopTicks);
+    const float kInputPoint =
+        static_cast<float>(std::round((kRevolutions * SECONDSTOMINUTE * MILISECONDSTOSECOND) / kTimeDt)) *
+        kSpeedIndexMultipler;
+    actualRpm_ = static_cast<int>(kInputPoint);
```

Comment:
1. Please extract method.
2. SECONDSTOMINUTE and MILISECONDSTOSECOND are not Google code style